calyx-opt/src/passes/compile_static.rs

```diff
@@ -346,6 +421,70 @@ impl Visitor for CompileStatic {
         Ok(Action::Change(Box::new(e)))
     }
 
+    fn start_while(
+        &mut self,
+        s: &mut ir::While,
+        comp: &mut ir::Component,
+        sigs: &ir::LibrarySignatures,
+        _comps: &[ir::Component],
+    ) -> VisResult {
+        // if while body is static, then we want to make sure that the while
+        // body does not take the extra cycle incurred by the done condition
+        // So we replace the while loop with `enable` of a wrapper group
+        // that sets the go signal of the static group in the while loop body high
+        // (all static control should be compiled into static groups by
+        // `static_inliner` now). The done signal of the wrapper group should be
+        // the condition that the fsm of the while body is %0 and the port signal
+        // is 1'd0.
+        // For example, we replace
+        // wires {
```

Comment:
Make this a documentation comment for the function (using `///`) and wrap the code examples with ticks around them to distinguish them from the text